state/futures/selectors.ts

```diff
@@ -102,15 +96,44 @@ export const selectIsolatedPriceImpact = createSelector(
 	(priceImpact) => wei(priceImpact, 0)
 );
 
-export const selectPerpsMarketRate = createSelector(selectMarketInfo, (marketInfo) => {
-	return marketInfo?.price ?? wei(0);
-});
+export const selectOrderType = createSelector(
+	(state: RootState) => state.futures,
+	(futures) => futures[accountType(futures.selectedType)].orderType
+);
 
-export const selectMarketAssetRate = createSelector(
-	(state: RootState) => state.futures[accountType(state.futures.selectedType)].selectedMarketAsset,
-	selectExchangeRates,
-	(marketAsset, exchangeRates) => {
-		return newGetExchangeRatesForCurrencies(exchangeRates, marketAsset, 'sUSD');
+export const selectMarketPrice = createSelector(
+	selectMarketAsset,
+	selectPrices,
+	selectOrderType,
+	(marketAsset, prices, orderType) => {
+		const price = prices[marketAsset];
+		return orderType === 'delayed offchain' ? price?.offChain ?? wei(0) : price?.onChain ?? wei(0);
+	}
+);
+
+export const selectLatestMarketPrice = createSelector(
+	selectMarketAsset,
+	selectPrices,
+	(marketAsset, prices) => {
+		const price = prices[marketAsset];
+		return price?.offChain ?? price?.onChain ?? wei(0);
+	}
+);
```

Comment:
Should this include order type in the same way `selectSkewAdjustedPrice` does? When market/delayed orders are selected, I think we want the onchain value.

Reply:
Maybe when we disable everything but delayed offchain it doesn't matter, just noting here.

Reply:
This is just if you don't care what price, you just want the latest for display, either off chain as priority or fallback to on chain